File: pininputlayout-demo/src/main/java/io/github/hendraanggrian/pininputlayoutdemo/MainActivity.java
package io.github.hendraanggrian.pininputlayoutdemo;

import android.os.Bundle;
import android.support.annotation.NonNull;
import android.support.v7.app.AppCompatActivity;
import android.widget.TextView;

import io.github.hendraanggrian.pininputlayout.PinInputLayout;

public class MainActivity extends AppCompatActivity {

    private PinInputLayout pinInputLayout;
    private TextView textViewState;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        textViewState = (TextView) findViewById(R.id.textview_state);

        pinInputLayout = (PinInputLayout) findViewById(R.id.pininputlayout);
        pinInputLayout.setOnStateChangedListener(new PinInputLayout.OnStateChangedListener() {
            @Override
            public void onStateChanged(@NonNull PinInputLayout view, @NonNull PinInputLayout.State state) {
                textViewState.setText(state.toString());
            }
        });
    }
}

File: example/src/com/example/pininputlayout/MainActivity.kt
package com.example.pininputlayout

import android.os.Bundle
import android.support.v7.app.AppCompatActivity
import android.view.View
import com.hendraanggrian.app.PinDialog
import kota.toast
import kotlinx.android.synthetic.main.activity_main.*

class MainActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        setSupportActionBar(toolbar)
        pinInputLayout.setOnStateChangedListener { _, isComplete -> textView.text = isComplete.toString() }
    }

    fun buttonOnClick(view: View) = PinDialog.Builder(this)
            .setTitle("Enter pin")
            .setNegativeButton(getText(android.R.string.cancel))
            .setPositiveButton(getText(android.R.string.ok), { _, _, text -> toast(text) })
            .build()
            .show()
}